### PR TITLE
add mypy pre-commit hook

### PR DESCRIPTION
**Pull request by bryevdv**
This PR adds a [pre-commit hook for mypy](https://github.com/pre-commit/mirrors-mypy). It should run `mypy legate` whenever there are relevant file changes.

### CODE REVIEW

**Review by manopapad — APPROVED** [review]
Works for me!

File: legate/core/corelib.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# Copyright 2021-2022 NVIDIA Corporation
 #
```
Comment (manopapad) [inline legate/core/corelib.py:1]: I assume we still want a copyright header?

Comment (bryevdv) [reply legate/core/corelib.py:1]: Oh weird, not sure why/how that happened

Comment (bryevdv) [inline legate/core/corelib.py:1]:
```suggestion
# Copyright 2021-2022 NVIDIA Corporation
```

Comment (bryevdv) [reply legate/core/corelib.py]: fixed (two commits, tried to be to clever with GH web ui suggestion)

**Pull request merged**